src/nvim/ex_cmds.c

```diff
@@ -3507,24 +3537,26 @@ void do_sub(exarg_T *eap)
          * use "\=col("."). */
         curwin->w_cursor.col = regmatch.startpos[0].col;
 
-        /*
-         * 3. substitute the string.
-         */
-        if (do_count) {
-          /* prevent accidentally changing the buffer by a function */
-          save_ma = curbuf->b_p_ma;
-          curbuf->b_p_ma = FALSE;
-          sandbox++;
-        }
-        /* get length of substitution part */
-        sublen = vim_regsub_multi(&regmatch,
-            sub_firstlnum - regmatch.startpos[0].lnum,
-            sub, sub_firstline, FALSE, p_magic, TRUE);
-        if (do_count) {
-          curbuf->b_p_ma = save_ma;
-          sandbox--;
-          goto skip;
-        }
+        //
+        // 3. substitute the string. don't do this while incsubstitution and
+        //    there's no word to replace by eg : ":%s/pattern"
+        //
```

Comment:
Empty `//` lines should not be present. Also sentenses start with capital letter.